### ISSUE TITLE
GAE.py中进行模型的训练和测试的时候为什么没有设置model.train(), model.eval()?

### ISSUE THREAD

**Issue by Arya-bioinformatics** [zyx423/GAE_VGAE#3]

**Comment by zyx423** [comment]
1、我当时写代码时候没写，你可以加上试试。 2、这个是无监督的，直接在训练时候进行的测试。取最好的结果。------------------&nbsp;原始邮件&nbsp;------------------
***@***.***&gt;
发送时间:&nbsp;2021年8月16日(星期一) 下午4:07
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[zyx423/GAE_VGAE] GAE.py中进行模型的训练和测试的时候为什么没有设置model.train(), model.eval()? (#3)

**Comment by Arya-bioinformatics** [comment]
感谢及时回复～
link prediction，用已知的adj和embedding的内积预测得到的adj做bce loss，应该算是有监督的吧？

**Comment by zyx423** [comment]
不是，还是无监督。 因为adj和特征矩阵一样，只是一个输入数据。重构的adj做loss只是无监督的一种手段，类似矩阵分解。 监督方法的训练过程中，是不仅有adj，还有每个节点的标签Y，gae里面没有Y。------------------&nbsp;原始邮件&nbsp;------------------
***@***.***&gt;
发送时间:&nbsp;2021年8月16日(星期一) 下午4:39
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [zyx423/GAE_VGAE] GAE.py中进行模型的训练和测试的时候为什么没有设置model.train(), model.eval()? (#3)
